code/game/machinery/autodoc_excelsior.dm

```diff
@@ -0,0 +1,249 @@
+/obj/machinery/excelsior_autodoc
+	name = "Excelsior Autodoc"
+	desc = "Medical care for everybody, free, and may no one be left behind!"
+	icon = 'icons/obj/machines/excelsior/autodoc.dmi'
+	icon_state = "base"
+	density = TRUE
+	anchored = TRUE
+	circuit = /obj/item/electronics/circuitboard/excelsior_autodoc
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 60
+	active_power_usage = 10000
+	var/mob/living/carbon/occupant
+	var/datum/autodoc/autodoc_processor
+	var/cover_closed = FALSE
+	var/cover_locked = FALSE
+	var/cover_moving = FALSE
+	var/image/screen_state = null
+	var/image/cover_state = null
+
+/obj/machinery/excelsior_autodoc/New()
+	. = ..()
+	autodoc_processor = new/datum/autodoc()
+	autodoc_processor.holder = src
+	autodoc_processor.damage_heal_amount = 20
+	update_icon()
+	
+
+/obj/machinery/excelsior_autodoc/Destroy()
+	if(occupant)
+		qdel(occupant)
+	return ..()
+
+/obj/machinery/excelsior_autodoc/relaymove(mob/user)
+	if (user.stat)
+		return
+	src.go_out()
+	return
+
+/obj/machinery/excelsior_autodoc/attackby(obj/item/I, mob/living/user)
+	if(default_deconstruction(I, user))
+		return
+	if(default_part_replacement(I, user))
+		return
+	..()
+
+/obj/machinery/excelsior_autodoc/verb/eject()
+	set src in view(1)
+	set category = "Object"
+	set name = "Eject Occupant"
+
+	if (usr.incapacitated())
+		return
+	src.go_out()
+	add_fingerprint(usr)
+	return
+
+/obj/machinery/excelsior_autodoc/verb/move_inside()
+	set src in view(1)
+	set category = "Object"
+	set name = "Enter Autodoc"
+
+	if(usr.stat)
```

Comment:
why not use incapacitated here ? since it checks for everything.

Reply:
Because that's how other machinery doing it.

Reply:
well they're doing it wrong by the looks of it then.